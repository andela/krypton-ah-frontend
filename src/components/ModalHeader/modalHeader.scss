@import "../colors";
@import "../fonts";

h2 {
  &.header {
    border-bottom: 0;
    color: lighten($primarycolor, 53%);
    
    font: {
      family: "roboto";
      size: 1.5rem;
      weight: 400;
    }
    margin-top: 1.25rem;
    text-transform: uppercase;
  }
}

hr {
  border: 1px solid;
  color: $black;
  margin: {
    bottom: 3.875rem;
    left: 0;
    top: -0.437rem;
  }
  width: 3.2rem;
}
