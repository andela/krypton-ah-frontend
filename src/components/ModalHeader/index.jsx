import React from 'react';
import { Modal } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import './modalHeader.scss';

function ModalHeader(props) {
  const { title } = props;
  return (
    <div>
      <Modal.Header as="h2">{ title }</Modal.Header>
      <hr />
    </div>
  );
}

ModalHeader.propTypes = {
  title: PropTypes.string
};
ModalHeader.defaultProps = {
  title: 'Sign Up'
};

export default ModalHeader;
