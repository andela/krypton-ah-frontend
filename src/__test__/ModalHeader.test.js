import React from 'react';
import { create } from 'react-test-renderer';
import checkPropTypes from 'check-prop-types';
import ModalHeader from '../components/ModalHeader';

function checkHeader() {
  test('it should matches the snapshot', () => {
    const component = create(<ModalHeader />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  test('Render title for Modal Header', () => {
    const expectedProps = { title: 'false' };
    const propError = checkPropTypes(ModalHeader.title, expectedProps, 'prop', ModalHeader.title);
    expect(propError).toBeUndefined();
  });
}
describe('Modal Header', () => {
  checkHeader();
});
