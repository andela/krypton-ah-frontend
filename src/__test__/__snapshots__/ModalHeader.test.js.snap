// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Header it should matches the snapshot 1`] = `
<div>
  <h2
    className="header"
  >
    Sign Up
  </h2>
  <hr />
</div>
`;
